src/GitHub.App/Controllers/UIController.cs

```diff
@@ -236,6 +236,14 @@ public void Start([AllowNull] IConnection connection)
                     .IsLoggedIn(hosts)
                     .Do(loggedin =>
                     {
+                        if (!loggedin && currentFlow != UIControllerFlow.Authentication)
+                        {
+                            connectionManager.Connections.CollectionChanged += (s, e) => {
```

Comment:
I need to test to make sure, but I believe this subscription gets killed when this instance dies (when the flow is done) and doesn't keep anything alive that shouldn't be. I'll check.